### PR TITLE
refactor: avoid overflow by explicitly casting operand to a wider type

### PR DESCRIPTION
**Pull request by dundargoc**

### CODE REVIEW

File: src/nvim/ex_docmd.c [hunk 1]
```diff
@@ -4370,7 +4370,7 @@ static char_u *replace_makeprg(exarg_T *eap, char_u *p, char_u **cmdlinep)
         ++i;
       }
       len = (int)STRLEN(p);
-      new_cmdline = xmalloc(STRLEN(program) + i * (len - 2) + 1);
+      new_cmdline = xmalloc(STRLEN(program) + (long)i * (len - 2) + 1);
```
Comment (bfredl) [inline src/nvim/ex_docmd.c:4373]: `size_t` ?

File: src/nvim/memfile.c [hunk 2]
```diff
@@ -269,7 +269,7 @@ bhdr_T *mf_new(memfile_T *mfp, bool negative, unsigned page_count)
 
   // Init the data to all zero, to avoid reading uninitialized data.
   // This also avoids that the passwd file ends up in the swap file!
-  (void)memset(hp->bh_data, 0, mfp->mf_page_size * page_count);
+  (void)memset(hp->bh_data, 0, (unsigned long)mfp->mf_page_size * page_count);
```
Comment (bfredl) [inline src/nvim/memfile.c:272]: `size_t`

File: src/nvim/screen.c [hunk 3]
```diff
@@ -6762,7 +6762,7 @@ void grid_clear_line(ScreenGrid *grid, unsigned off, int width, bool valid)
 
 void grid_invalidate(ScreenGrid *grid)
 {
-  (void)memset(grid->attrs, -1, grid->Rows * grid->Columns * sizeof(sattr_T));
+  (void)memset(grid->attrs, -1, (long)grid->Rows * grid->Columns * sizeof(sattr_T));
```
Comment (bfredl) [inline src/nvim/screen.c:6765]: `size_t`

Comment (zeertzjq) [reply src/nvim/screen.c:6765]: Is this necessary?

File: src/nvim/ex_docmd.c (hunk 1 above)
Comment (zeertzjq) [reply src/nvim/ex_docmd.c:4373]: Is this even necessary?

File: src/nvim/screen.c (hunk 3 above)
Comment (zeertzjq) [reply src/nvim/screen.c:6765]:
```suggestion
  (void)memset(grid->attrs, -1, sizeof(sattr_T) * grid->Rows * grid->Columns);
```

Comment (zeertzjq) [inline src/nvim/screen.c:6765]:
```suggestion
  (void)memset(grid->attrs, -1, sizeof(sattr_T) * grid->Rows * grid->Columns);
```

**Review by zeertzjq — APPROVED** [review]

**Pull request merged**